packages/service-provider-server/src/cli-service-provider.integration.spec.ts

```diff
@@ -661,26 +661,6 @@ describe('CliServiceProvider [integration]', function() {
     });
   });
 
-  describe('db fetching', () => {
-    it('returns the same db instance when used with the same name and options', () => {
-      const db1 = serviceProvider._dbTestWrapper('foo', { readPreference: { mode: 'secondary' } } as DbOptions);
-      const db2 = serviceProvider._dbTestWrapper('foo', { readPreference: { mode: 'secondary' } } as DbOptions);
-      expect(db1).to.equal(db2);
-    });
-
-    it('returns the different db instances when used with different names', () => {
-      const db1 = serviceProvider._dbTestWrapper('bar', { readPreference: { mode: 'secondary' } } as DbOptions);
-      const db2 = serviceProvider._dbTestWrapper('foo', { readPreference: { mode: 'secondary' } } as DbOptions);
-      expect(db1).not.to.equal(db2);
-    });
-
-    it('returns the different db instances when used with different options', () => {
-      const db1 = serviceProvider._dbTestWrapper('foo', { readPreference: { mode: 'primary' } } as DbOptions);
-      const db2 = serviceProvider._dbTestWrapper('foo', { readPreference: { mode: 'secondary' } } as DbOptions);
-      expect(db1).not.to.equal(db2);
-    });
-  });
```

Comment:
I can see that they were failing in CI, but that surprises me, because these are not supposed to be affected by the driver’s caching at all and instead are supposed to test our *own* caching, and that the tests failed because `db()` returned the same `Db` instance for different options does seem a bit concerning to me … do you know more about this?

Reply:
It looks like for now the caching feature is still alive and well in the driver’s master branch: https://github.com/mongodb/node-mongodb-native/blob/cb9ee9e6175a6654c3c300801884e4a3c3a653ac/src/mongo_client.ts#L404

Reply:
Ah, it seems this is in code review https://jira.mongodb.org/browse/NODE-2931 so not quite in master yet, so that was a little presumtious of me to make this PR